### PR TITLE
Add /git_no_unix switch to init.bat to prevent adding !GIT_INSTALL_ROOT!\usr\bin to PATH

### PR DESCRIPTION
**Pull request by patstew**
Adding all the unix tools to the path prevents some programs (Vivado) from working for me.

### CODE REVIEW

**Comment by daxgames** [comment]
I'm not sure about this feature.  If you don't want *nix tools why not use `cmder_mini`.

**Comment by daxgames** [comment]
or remove the unix tools from the path using youe `user_profile.cmd`

**Comment by patstew** [comment]
I am using cmder_mini, but cmder still picks up my git for windows installation. I like having the git prompt integration etc, and want 'git' on the path, but having 'sh' _anywhere_ on the path causes problems for me. I'd rather just maintain my own init.bat with that line commented than write a script to edit the PATH afterwards. Git for windows offers 2 options for path integration when you install, with and without the unix tools, I assume because they know it can be a problem (not just one that affects me).

**Comment by daxgames** [comment]
I understand.  I am working on a switch to `/prefer_nix` that adds `*nix` commands to the front of the path vs. the default of adding to the end of the path.  This is how `Cmder` used to setup the path and users have requested `*nix` tool preference.

FWIW you can do this today with existing conditional call features in Cmder and `init.bat`:  

1. Clone a Conemu Cmder task, call it `Cmder No Unix Tools`
2. Add `/no_nix` to the command line.
3. Create a `%cmder_root%\bin\no_nix.bat` file with:

```
set path=%PATH:C:\Users\user\cmderdev\vendor\git-for-windows\usr\bin;=%
```

4. Edit `%cmder_root%\config\user_profile.cmd` and add.
```
%ccall% /no_nix no_nix
```

**Comment by daxgames** [comment]
@patstew A clarification.  I just want to make sure your change works with mine before merging.

File: vendor/init.bat [hunk 1]
```diff
@@ -239,7 +242,9 @@ if defined GIT_INSTALL_ROOT (
     ) else if exist "!GIT_INSTALL_ROOT!\mingw64" (
         %lib_path% enhance_path "!GIT_INSTALL_ROOT!\mingw64\bin" append
     )
-    %lib_path% enhance_path "!GIT_INSTALL_ROOT!\usr\bin" append
+    if %git_unix_path% gtr 0 (
```
Comment (daxgames) [inline vendor/init.bat]: Change to:

```
if "%nix_tools%" == "0" (
```

File: vendor/init.bat [hunk 2]
```diff
@@ -79,6 +80,8 @@ call "%cmder_root%\vendor\lib\lib_profile"
             %lib_console% show_error "The Git install root folder "%~2", you specified does not exist!"
             exit /b
         )
+    ) else if /i "%1"=="/git_no_unix" (
+        set git_unix_path=0
```
Comment (daxgames) [inline vendor/init.bat]: Delete line 84

**Comment by daxgames** [comment]
👍 LGTM

**Comment by daxgames** [comment]
@Stanzilla - You want to look.  I tested it and it works.

**Comment by Stanzilla** [comment]
LGTM!

**Comment by daxgames** [comment]
@patstew you have any issues with #2222 that implements `/nix_tools 0` should prevent adding !GIT_INSTALL_ROOT!\mingw64\bin to PATH #2214

**Pull request merged**